### PR TITLE
Support swagger version 2

### PR DESCRIPTION
**Pull request by Tharsanan1**

### CODE REVIEW

**Review by rosensilva — APPROVED** [review]

File: components/esb-tools/plugins/org.wso2.integrationstudio.artifact.synapse.api/src/org/wso2/integrationstudio/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java [hunk 1]
```diff
@@ -147,6 +161,12 @@ public class SynapseAPICreationWizard extends AbstractWSO2ProjectCreationWizard
     private static final String WHITE_SPACE = " ";
     private static final String METADATA_TYPE = "synapse/metadata";
     private static final String REGISTRY_RESOURCE_PATH = "/_system/governance/swagger_files";
+    private static final String SWAGGER_CONVERTER_API_URL = "https://converter.swagger.io/api/convert";
+    private static final String SWAGGER_API_CALL_CONSENT_TITLE = "Convert to Swagger version 3?";
+    private static final String SWAGGER_API_CALL_CONSENT_MESSAGE = "Integration Studio supports Swagger version 3 and above only. The Swagger file you have uploaded is version 2. You can convert this file to version 3 by clicking “Yes”. Note that Integration Studio uses the https://converter.swagger.io API to convert your payload.\n\nYou can continue with the Swagger file you uploaded by clicking “No”, if you feel the detected version is incorrect.\n\nClick “Cancel” to terminate the API creation process.";
```
Comment (lasanthaS) [inline components/esb-tools/plugins/org.wso2.integrationstudio.artifact.synapse.api/src/org/wso2/integrationstudio/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java:166]: Break the line at 120 length

File: components/esb-tools/plugins/org.wso2.integrationstudio.artifact.synapse.api/src/org/wso2/integrationstudio/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java [hunk 2]
```diff
@@ -967,6 +1030,15 @@ public void openEditor(File file) {
             log.error("Cannot open the editor", e);
         }
     }
+    
+    private boolean isVersionTwoYamlFile(String yamlContent) {
+    	Pattern p = Pattern.compile("([ ]*)(swagger[ ]*:[ ]*[\\\"]?[2][.])(.*)");
+    	Matcher m = p.matcher(yamlContent.toLowerCase());
+    	if (m.find()) {
+    		return true;
```
Comment (shilmyhasan) [inline components/esb-tools/plugins/org.wso2.integrationstudio.artifact.synapse.api/src/org/wso2/integrationstudio/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java:1038]: formatting issue

File: components/esb-tools/plugins/org.wso2.integrationstudio.artifact.synapse.api/src/org/wso2/integrationstudio/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java [hunk 3]
```diff
@@ -750,18 +770,61 @@ public static String convertJSONtoYaml(String jsonSource) throws Exception {
...
     private String getSwaggerFileAsYAML(File swaggerFile, String apiName) {
         String swaggerContent = "";
+        boolean terminateFlag = false;
         try {
             swaggerContent = new String(Files.readAllBytes(Paths.get(swaggerFile.getAbsolutePath())));
             if (FilenameUtils.getExtension(swaggerFile.getAbsolutePath()).equals("json")) {
                 swaggerContent = convertJSONtoYaml(swaggerContent);
             }
+            if (isVersionTwoYamlFile(swaggerContent)) {
+	        	Shell shell = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell();;
```
Comment (shilmyhasan) [inline components/esb-tools/plugins/org.wso2.integrationstudio.artifact.synapse.api/src/org/wso2/integrationstudio/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java:780]: appears a formatting issue, better use wso2 formatting template

File: components/esb-tools/plugins/org.wso2.integrationstudio.artifact.synapse.api/src/org/wso2/integrationstudio/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java (hunk 1 above)
Comment (Tharsanan1) [reply components/esb-tools/plugins/org.wso2.integrationstudio.artifact.synapse.api/src/org/wso2/integrationstudio/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java:166]: This PR is reverted by: https://github.com/Tharsanan1/integration-studio/commit/ef1e82bf7abe20bfb8300d844d2533ca4ffbb6dd. Hence resolving the comment

File: components/esb-tools/plugins/org.wso2.integrationstudio.artifact.synapse.api/src/org/wso2/integrationstudio/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java (hunk 3 above)
Comment (Tharsanan1) [reply components/esb-tools/plugins/org.wso2.integrationstudio.artifact.synapse.api/src/org/wso2/integrationstudio/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java:780]: This PR is reverted by: https://github.com/Tharsanan1/integration-studio/commit/ef1e82bf7abe20bfb8300d844d2533ca4ffbb6dd. Hence resolving the comment

File: components/esb-tools/plugins/org.wso2.integrationstudio.artifact.synapse.api/src/org/wso2/integrationstudio/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java (hunk 2 above)
Comment (Tharsanan1) [reply components/esb-tools/plugins/org.wso2.integrationstudio.artifact.synapse.api/src/org/wso2/integrationstudio/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java:1038]: This PR is reverted by: https://github.com/Tharsanan1/integration-studio/commit/ef1e82bf7abe20bfb8300d844d2533ca4ffbb6dd. Hence resolving the comment

**Pull request merged**